### PR TITLE
Build Base Flutter - PART 1

### PR DESCRIPTION
**Pull request by chuongdevJC**
- Config Dio (like okHttp), interceptor
- Repository
- Flutter BloC
- Injection
- Demo list user

### CODE REVIEW

File: lib/bloc/blocs/user_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        if (currentState is UserGitUnInitialized) {
+          final users = await fetchUserGit(1);
+          yield UserGitLoaded(users: users, hasReachMax: false);
+        } else if (currentState is UserGitLoaded) {
+          var page =
+              ((currentState as UserGitLoaded).users.length / 10).ceil() + 1;
+          final users = await fetchUserGit(page);
+          yield users.length < 10
+              ? (currentState as UserGitLoaded).copyWith(hasReachedMax: true)
+              : UserGitLoaded(
```
Comment (vuphu-jc) [inline lib/bloc/blocs/user_bloc.dart:42]: I think we should save variable '_page', initial value is 1 and plus one after each fetch data

Comment (chuongdevJC) [reply lib/bloc/blocs/user_bloc.dart:42]: Actually we dont need count when we fetch data. make it easy from data we have. it's flexible and beside we have a variable 'isReachMax' so we never load wrong data. BTW this is just demo.

Comment (vuphu-jc) [reply lib/bloc/blocs/user_bloc.dart:42]: Yep, I totally agree with you!

File: lib/repositories/user_repository.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  final DioClient _dioClient;
+
+  UserRepositoryImpl(this._dioClient);
+
+  @override
+  Future<List<UserGit>> getUser(int page) async {
+    final response = await _dioClient.get('/search/users',
+        queryParameters: {'q': 'abc', 'page': page, 'per_page': 10});
+    var tagObjsJson = jsonDecode(json.encode(response))['items'] as List;
```
Comment (daolq3012) [inline lib/repositories/user_repository.dart:25]: .

**Review by daolq3012 — CHANGES_REQUESTED** [review]
.

**Review by lamlejobchat — APPROVED** [review]
LGTM!

**Pull request merged**